p2p/discovery/bucket.go

```diff
@@ -0,0 +1,53 @@
+/**
+*  @file
+*  @copyright defined in go-seele/LICENSE
+ */
+package discovery
+
+import (
+	"github.com/seeleteam/go-seele/log"
+)
+
+const (
+	bucketSize = 16 // Kademlia bucket size
+)
+
+type bucket struct {
+	peers []*Node
+}
+
+func NewBuckets() *bucket {
+	return &bucket{
+		peers: make([]*Node, 0),
+	}
+}
+
+// addNode add node to bucket, if bucket is full, will remove an old one
+func (b *bucket) addNode(node *Node) {
+	index := b.hasNode(node)
+
+	if index != -1 {
+		// do nothing for now
+		// TODO lru
+	} else {
+		if len(b.peers) < bucketSize {
+			b.peers = append(b.peers, node)
+		} else {
+			copy(b.peers[:], b.peers[1:])
```

Comment:
just confirm, it's safe to copy within the same array?

Reply:
yes